lib/WebService/Braintree/Gateway.pm

```diff
@@ -3,108 +3,42 @@ package WebService::Braintree::Gateway;
-});
-
-has 'customer' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::CustomerGateway->new(gateway => $self);
-});
-
-has 'merchant_account' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::MerchantAccountGateway->new(gateway => $self);
-});
-
-has 'payment_method' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::PaymentMethodGateway->new(gateway => $self);
-});
-
-has 'payment_method_nonce' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::PaymentMethodNonceGateway->new(gateway => $self);
-});
-
-has 'paypal_account' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::PayPalAccountGateway->new(gateway => $self);
-});
-
-has 'plan' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::PlanGateway->new(gateway => $self);
-});
-
-has 'settlement_batch_summary' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::SettlementBatchSummaryGateway->new(gateway => $self);
-});
-
-has 'subscription' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::SubscriptionGateway->new(gateway => $self);
-});
-
-has 'transaction' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::TransactionGateway->new(gateway => $self);
-});
-
-has 'transparent_redirect' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::TransparentRedirectGateway->new(gateway => $self);
-});
-
-has 'webhook_notification' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::WebhookNotificationGateway->new(gateway => $self);
-});
-
-has 'webhook_testing' => (is => 'ro', lazy => 1, default => sub {
-    my $self = shift;
-    WebService::Braintree::WebhookTestingGateway->new(gateway => $self);
-});
+my %gateways = (
+  add_on => 'AddOn',
+  address => 'Address',
+  apple_pay => 'ApplePay',
+  client_token => 'ClientToken',
+  credit_card => 'CreditCard',
+  credit_card_verification => 'CreditCardVerification',
+  customer => 'Customer',
+  discount => 'Discount',
+  dispute => 'Dispute',
+  merchant => 'Merchant',
+  merchant_account => 'MerchantAccount',
+  payment_method => 'PaymentMethod',
+  payment_method_nonce => 'PaymentMethodNonce',
+  paypal_account => 'PayPalAccount',
+  plan => 'Plan',
+  settlement_batch_summary => 'SettlementBatchSummary',
+  subscription => 'Subscription',
+  transaction => 'Transaction',
+  transparent_redirect => 'TransparentRedirect',
+  webhook_notification => 'WebhookNotification',
+  webhook_testing => 'WebhookTesting',
+);
+
+while (my ($method, $gateway) = each %gateways) {
+  my $package = "WebService::Braintree::${gateway}Gateway";
+  eval "use $package"; die $@ if $@;
```

Comment:
not hugely fond of the string eval here. Moose provides Class::Load::XS so I've made a diff and tested it (see end of review)